base/linalg/arnoldi.jl

```diff
@@ -237,7 +237,11 @@ function _eigs(A, B;
     end
 
     # Refer to ex-*.doc files in ARPACK/DOCUMENTS for calling sequence
-    matvecA(x) = A * x
+    if method_exists(Base.A_mul_B!, (Vector{T},typeof(A),Vector{T}))
```

Comment:
Probably much cleaner to just replace uses of `matvecA` with `A_mul_B!` here. Using `method_exists` is going to be slow and difficult to predict.